doc/usage/spark-demo-tutorial.md

```diff
@@ -0,0 +1,168 @@
+kind: AppWrapper
+metadata:
+  name: spark-c3-sx
+  labels:
+    quota_context: "context-3"
+    quota_service: "service-x"
+spec:
+  resources:
+    GenericItems:
+      - replicas: 1
+        generictemplate:
+          apiVersion: v1
+          kind: ReplicationController
+          metadata:
+            name: spark-master-controller
+          spec:
+            replicas: 1
+            selector:
+              component: spark-master
+            template:
+              metadata:
+                labels:
+                  component: spark-master
+              spec:
+                containers:
+                  - name: spark-master
+                    image: k8s.gcr.io/spark:1.5.2_v1
+                    command: ["/start-master"]
+                    ports:
+                      - containerPort: 7077
+                      - containerPort: 8080
+                    resources:
+                      requests:
+                        cpu: 100m
+      - replicas: 1
+        generictemplate:
+          apiVersion: v1
+          kind: ReplicationController
+          metadata:
+            name: spark-worker-controller
+          spec:
+            replicas: 3
+            selector:
+              component: spark-worker
+            template:
+              metadata:
+                labels:
+                  component: spark-worker
+              spec:
+                containers:
+                  - name: spark-worker
+                    image: k8s.gcr.io/spark:1.5.2_v1
+                    command: ["/start-worker"]
+                    ports:
+                      - containerPort: 8081
+                    resources:
+                      requests:
+                        cpu: 10m
+      - replicas: 1
+        generictemplate:
+          apiVersion: batch/v1
+          kind: Job
+          metadata:
+            name: spark-job
+          spec:
+            template:
+              spec:
+                containers:
+                  - name: spark-job
+                    image: k8s.gcr.io/spark:1.5.2_v1
+                    command: ["/bin/bash", "-c"]
+                    args: ["/opt/spark/bin/spark-submit --master spark://spark-master:7077 /opt/spark/examples/src/main/python/pi.py"]
+                    ports:
+                      - containerPort: 8080
+                restartPolicy: OnFailure
+            backoffLimit: 2
+      - replicas: 1
+        generictemplate:
+          apiVersion: v1
+          kind: Service
+          metadata:
+            name: spark-master
+          spec:
+            ports:
+              - port: 7077
+                targetPort: 7077
+                name: spark
+              - port: 8080
+                targetPort: 8080
+                name: http
+            selector:
+              component: spark-master
+
+```
+
+### Running a Spark Job
+
+Once the AppWrapper job is created. It can be used to create pods to run Spark Master Services, Spark Workers and Spark Job. 
```

Comment:
Suggestion: "Once the AppWrapper job is created. It can be used to create pods..." to "Once the MCAD Controller dispatches the AppWrapper the MCAD controller will then create the items listed in the AppWrapper resulting in the creation of pods for the Master Services, Spark Workers and Spark Job will be created."   Trying to make it clear that the MCAD handles all the steps above and clarifying that the MCAD Controller does not create pods in the example YAML but instead create the objects (Replication Controllers) and some other controller outside the scope of MCAD (in this case the Kubernetes Controller) will create the pods.